.cargo/config.toml

```diff
@@ -0,0 +1,2 @@
+[build]
+rustflags = ["-Dwarnings", "-Ctarget-feature=+sse4.1,+sse4.2"]
```

Comment:
I think having `-Dwarnings` here will mean that during development there cannot be any warnings at any time, otherwise the compilation will fail. In my experience this is not a deal breaker, but having to keep codebase warning-free at all times definitely makes development way less straightforward.

Reply:
Either we do this or having GNU make syntax (which works for Mac BTW).

Reply:
We could plausibly not set `-Dwarnings` in either place. The most appropriate place for `-Dwarnings` would be in some CI job, probably.

Reply:
And yeah if we have it this way it won't compile on M1 processors because sse4.1 warnings will turn into errors. I would propose to go with having it in Makefile.

Reply:
`-Dwarnings` in the Makefile would still mean being unable to use the Makefile when building for aarch64.

Reply:
It won't if we enable sse4.1 only for x86_64:
```ifeq ($(shell uname -m),x86_64)
  export RUSTFLAGS := ${RUSTFLAGS} -Ctarget-feature=+sse4.1,+sse4.2
endif```

Reply:
```
[build.rustflags](https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags)
Type: string or array of strings
Default: none
Environment: CARGO_BUILD_RUSTFLAGS or RUSTFLAGS
Extra command-line flags to pass to rustc. The value may be a array of strings or a space-separated string.

There are three mutually exclusive sources of extra flags. They are checked in order, with the first one being used:

RUSTFLAGS environment variable.
All matching target.<triple>.rustflags and target.<cfg>.rustflags config entries joined together.
build.rustflags config value.
```